# London and New York Similar Neighbourhoods

Load Libraries

In [1]:
import requests
from bs4 import BeautifulSoup
import pandas as pd
import numpy as np
import json

## Tube Stations in London

Use open data from TFL and convert kml into geojson https://api-portal.tfl.gov.uk/docs

In [2]:
import kml2geojson
kml2geojson.main.convert('Data/Londonstations.kml', 'Data/', separate_folders=False, style_type=None, style_filename='style.json')

Load the geojson 

In [3]:
with open('Data/Londonstations.geojson') as json_data:
    london_data = json.load(json_data)

Review the structure of the geojson

In [4]:
stations_data = london_data['features']
stations_data[0]

{'type': 'Feature',
 'properties': {'name': 'Acton Town Station',
  'description': 'Acton Town Station, London Underground Ltd., Gunnersbury Lane, London, W3 8HN',
  'styleUrl': '#tubeStyle'},
 'geometry': {'type': 'Point',
  'coordinates': [-0.2802512035361106, 51.50274977300057, 0.0]}}

Create a Pandas Dataframe

In [116]:
dataframe_columns = ['Name','City','Latitude','Longitude']
london_dataframe = pd.DataFrame(columns=dataframe_columns)

Iterate through the stations data to populate the Datafram

In [123]:
for data in stations_data:
    station_name = data['properties']['name']
    
    station_latlon = data['geometry']['coordinates']
    station_lat = station_latlon[1]
    station_lon = station_latlon[0]

    
    london_dataframe = london_dataframe.append({
                            'Name': station_name,
                            'City': 'London',
                            'Latitude': station_lat,
                            'Longitude': station_lon}, ignore_index=True)

In [124]:
london_dataframe.head()

Name    City   Latitude  Longitude
0    Acton Town Station  London  51.502750  -0.280251
1       Aldgate Station  London  51.514272  -0.075614
2  Aldgate East Station  London  51.515233  -0.072287
3      Alperton Station  London  51.540695  -0.299487
4      Amersham Station  London  51.674150  -0.607479

In [32]:
print('London has {} tube stations'.format(
        len(london_dataframe['Name']),
        london_dataframe.shape[0]
    )
)

London has 302 tube stations


## New York Neighbourhoods

In [10]:
!wget -q -O 'newyork_data.json' https://cocl.us/new_york_dataset
print('Data downloaded!')

Data downloaded!


In [11]:
with open('Data/newyork_data.json') as json_data:
    new_york_data = json.load(json_data)

In [12]:
neighborhoods_data = new_york_data['features']
neighborhoods_data[0]

{'type': 'Feature',
 'id': 'nyu_2451_34572.1',
 'geometry': {'type': 'Point',
  'coordinates': [-73.84720052054902, 40.89470517661]},
 'geometry_name': 'geom',
 'properties': {'name': 'Wakefield',
  'stacked': 1,
  'annoline1': 'Wakefield',
  'annoline2': None,
  'annoline3': None,
  'annoangle': 0.0,
  'borough': 'Bronx',
  'bbox': [-73.84720052054902,
   40.89470517661,
   -73.84720052054902,
   40.89470517661]}}

In [119]:
column_names = ['Name', 'City', 'Latitude', 'Longitude'] 
new_york_dataframe = pd.DataFrame(columns=column_names)

In [121]:
for data in neighborhoods_data:
    neighborhood_name = data['properties']['name']
        
    neighborhood_latlon = data['geometry']['coordinates']
    neighborhood_lat = neighborhood_latlon[1]
    neighborhood_lon = neighborhood_latlon[0]
    
    new_york_dataframe = new_york_dataframe.append({
                                'Name': neighborhood_name,
                                'City': 'New York',
                                'Latitude': neighborhood_lat,
                                'Longitude': neighborhood_lon}, ignore_index=True)

In [132]:
new_york_dataframe.head()

Name      City   Latitude  Longitude
0    Wakefield  New York  40.894705 -73.847201
1   Co-op City  New York  40.874294 -73.829939
2  Eastchester  New York  40.887556 -73.827806
3    Fieldston  New York  40.895437 -73.905643
4    Riverdale  New York  40.890834 -73.912585

Removing duplicates

In [168]:
new_york_dataframe.drop_duplicates(['Name'], keep=False, inplace=True)
new_york_dataframe.reset_index()

index                 Name      City   Latitude  Longitude
0        0            Wakefield  New York  40.894705 -73.847201
1        1           Co-op City  New York  40.874294 -73.829939
2        2          Eastchester  New York  40.887556 -73.827806
3        3            Fieldston  New York  40.895437 -73.905643
4        4            Riverdale  New York  40.890834 -73.912585
5        5          Kingsbridge  New York  40.881687 -73.902818
6        6          Marble Hill  New York  40.876551 -73.910660
7        7             Woodlawn  New York  40.898273 -73.867315
8        8              Norwood  New York  40.877224 -73.879391
9        9       Williamsbridge  New York  40.881039 -73.857446
10      10           Baychester  New York  40.866858 -73.835798
11      11       Pelham Parkway  New York  40.857413 -73.854756
12      12          City Island  New York  40.847247 -73.786488
13      13         Bedford Park  New York  40.870185 -73.885512
14      14   University Heights  New York  40.855727 -73.910416
15      15       Morris Heights  New York  40.847898 -73.919672
16      16              Fordham  New York  40.860997 -73.896427
17      17         East Tremont  New York  40.842696 -73.887356
18      18           West Farms  New York  40.839475 -73.877745
19      19         High  Bridge  New York  40.836623 -73.926102
20      20              Melrose  New York  40.819754 -73.909422
21      21           Mott Haven  New York  40.806239 -73.916100
22      22          Port Morris  New York  40.801664 -73.913221
23      23             Longwood  New York  40.815099 -73.895788
24      24          Hunts Point  New York  40.809730 -73.883315
25      25           Morrisania  New York  40.823592 -73.901506
26      26            Soundview  New York  40.821012 -73.865746
27      27         Clason Point  New York  40.806551 -73.854144
28      28          Throgs Neck  New York  40.815109 -73.816350
29      29         Country Club  New York  40.844246 -73.824099
..     ...                  ...       ...        ...        ...
268    276             Flatiron  New York  40.739673 -73.990947
269    277    Sunnyside Gardens  New York  40.745652 -73.918193
270    278           Blissville  New York  40.737251 -73.932442
271    279         Fulton Ferry  New York  40.703281 -73.995508
272    280         Vinegar Hill  New York  40.703321 -73.981116
273    281           Weeksville  New York  40.675040 -73.930531
274    282    Broadway Junction  New York  40.677861 -73.903317
275    283                Dumbo  New York  40.703176 -73.988753
276    284        Manor Heights  New York  40.601810 -74.120594
277    285          Willowbrook  New York  40.603707 -74.132084
278    286         Sandy Ground  New York  40.541140 -74.217766
279    287          Egbertville  New York  40.579119 -74.127272
280    288              Roxbury  New York  40.567376 -73.892138
281    289            Homecrest  New York  40.598525 -73.959185
282    290       Middle Village  New York  40.716415 -73.881143
283    291         Prince's Bay  New York  40.526264 -74.201526
284    292      Lighthouse Hill  New York  40.576506 -74.137927
285    293      Richmond Valley  New York  40.519541 -74.229571
286    294                Malba  New York  40.790602 -73.826678
287    295        Highland Park  New York  40.681999 -73.890346
288    296              Madison  New York  40.609378 -73.948415
289    297            Bronxdale  New York  40.852723 -73.861726
290    298             Allerton  New York  40.865788 -73.859319
291    299  Kingsbridge Heights  New York  40.870392 -73.901523
292    300              Erasmus  New York  40.646926 -73.948177
293    301         Hudson Yards  New York  40.756658 -74.000111
294    302              Hammels  New York  40.587338 -73.805530
295    303            Bayswater  New York  40.611322 -73.765968
296    304         Queensbridge  New York  40.756091 -73.945631
297    305            Fox Hills  New York  40.617311 -74.081740

[298 rows x 5 columns]

In [169]:
print('New York has {} Neighborhoods'.format(
        len(new_york_dataframe['Name']),
        new_york_dataframe.shape[0]
    )
)

New York has 298 Neighborhoods


### First interesting trivia: There are 302 Tube stations in London and 302 Neighborhoods in New York!

## New York and London as a massive super city across the Atlantic

In [170]:
super_city = london_dataframe.append(new_york_dataframe, ignore_index=True)

In [171]:
super_city

Name      City   Latitude  Longitude
0         Acton Town Station    London  51.502750  -0.280251
1            Aldgate Station    London  51.514272  -0.075614
2       Aldgate East Station    London  51.515233  -0.072287
3           Alperton Station    London  51.540695  -0.299487
4           Amersham Station    London  51.674150  -0.607479
5              Angel Station    London  51.532499  -0.105790
6            Archway Station    London  51.565427  -0.135114
7        Arnos Grove Station    London  51.616340  -0.133516
8            Arsenal Station    London  51.558499  -0.105762
9       Baker Street Station    London  51.523061  -0.156900
10            Balham Station    London  51.443329  -0.153208
11              Bank Station    London  51.513302  -0.088916
12          Barbican Station    London  51.520146  -0.097711
13           Barking Station    London  51.539451   0.080863
14       Barkingside Station    London  51.585786   0.088511
15      Barons Court Station    London  51.490160  -0.213607
16         Bayswater Station    London  51.512233  -0.188038
17         Becontree Station    London  51.540289   0.127400
18      Belsize Park Station    London  51.550427  -0.164232
19        Bermondsey Station    London  51.497949  -0.063651
20     Bethnal Green Station    London  51.527195  -0.055431
21       Blackfriars Station    London  51.511491  -0.103607
22   Blackhorse Road Station    London  51.586951  -0.040996
23       Bond Street Station    London  51.513805  -0.149332
24           Borough Station    London  51.501194  -0.093703
25      Boston Manor Station    London  51.495696  -0.324764
26      Bounds Green Station    London  51.607091  -0.124219
27          Bow Road Station    London  51.526800  -0.024824
28       Brent Cross Station    London  51.576800  -0.213456
29           Brixton Station    London  51.462581  -0.114636
..                       ...       ...        ...        ...
872                 Flatiron  New York  40.739673 -73.990947
873        Sunnyside Gardens  New York  40.745652 -73.918193
874               Blissville  New York  40.737251 -73.932442
875             Fulton Ferry  New York  40.703281 -73.995508
876             Vinegar Hill  New York  40.703321 -73.981116
877               Weeksville  New York  40.675040 -73.930531
878        Broadway Junction  New York  40.677861 -73.903317
879                    Dumbo  New York  40.703176 -73.988753
880            Manor Heights  New York  40.601810 -74.120594
881              Willowbrook  New York  40.603707 -74.132084
882             Sandy Ground  New York  40.541140 -74.217766
883              Egbertville  New York  40.579119 -74.127272
884                  Roxbury  New York  40.567376 -73.892138
885                Homecrest  New York  40.598525 -73.959185
886           Middle Village  New York  40.716415 -73.881143
887             Prince's Bay  New York  40.526264 -74.201526
888          Lighthouse Hill  New York  40.576506 -74.137927
889          Richmond Valley  New York  40.519541 -74.229571
890                    Malba  New York  40.790602 -73.826678
891            Highland Park  New York  40.681999 -73.890346
892                  Madison  New York  40.609378 -73.948415
893                Bronxdale  New York  40.852723 -73.861726
894                 Allerton  New York  40.865788 -73.859319
895      Kingsbridge Heights  New York  40.870392 -73.901523
896                  Erasmus  New York  40.646926 -73.948177
897             Hudson Yards  New York  40.756658 -74.000111
898                  Hammels  New York  40.587338 -73.805530
899                Bayswater  New York  40.611322 -73.765968
900             Queensbridge  New York  40.756091 -73.945631
901                Fox Hills  New York  40.617311 -74.081740

[902 rows x 4 columns]

Using Foursquare API

In [36]:
CLIENT_ID = 'GCX50K20UQTRSFVRADR2H0SNXLKVS5PMD0KLXOYSWFTNSTIL' 
CLIENT_SECRET = 'HIXGV1MRDYLMBBTAUKB5BKSH1PF4PMNAR0K3TB55PIJXTZQX' 
VERSION = '20180605' 
LIMIT = 200
radius = 1000

In [41]:
## GetNearbyVenues Function to Store Venues in a Neighbourhood
def getNearbyVenues(names, latitudes, longitudes, radius=1500):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        # make the GET request with a try/except construct to retry if errors occur
        try:
            results = requests.get(url, timeout=5).json()["response"]['groups'][0]['items']
        except requests.exceptions.Timeout:
            try:
                results = requests.get(url, timeout=5).json()["response"]['groups'][0]['items']
            except requests.exceptions.Timeout: 
                pass
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Neighborhood', 
                  'Neighborhood Latitude', 
                  'Neighborhood Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    return(nearby_venues)

super_city_venues = getNearbyVenues(names=super_city['Name'],
                                   latitudes=super_city['Latitude'],
                                   longitudes=super_city['Longitude']
                                  )

Wakefield
Co-op City
Eastchester
Fieldston
Riverdale
Kingsbridge
Marble Hill
Woodlawn
Norwood
Williamsbridge
Baychester
Pelham Parkway
City Island
Bedford Park
University Heights
Morris Heights
Fordham
East Tremont
West Farms
High  Bridge
Melrose
Mott Haven
Port Morris
Longwood
Hunts Point
Morrisania
Soundview
Clason Point
Throgs Neck
Country Club
Parkchester
Westchester Square
Van Nest
Morris Park
Belmont
Spuyten Duyvil
North Riverdale
Pelham Bay
Schuylerville
Edgewater Park
Castle Hill
Olinville
Pelham Gardens
Concourse
Unionport
Edenwald
Bay Ridge
Bensonhurst
Sunset Park
Greenpoint
Gravesend
Brighton Beach
Sheepshead Bay
Manhattan Terrace
Flatbush
Crown Heights
East Flatbush
Kensington
Windsor Terrace
Prospect Heights
Brownsville
Williamsburg
Bushwick
Bedford Stuyvesant
Brooklyn Heights
Cobble Hill
Carroll Gardens
Red Hook
Gowanus
Fort Greene
Park Slope
Cypress Hills
East New York
Starrett City
Canarsie
Flatlands
Mill Island
Manhattan Beach
Coney Island
Bath Beach
Borough Park
Dyker

In [42]:
super_city_venues.head()

Neighborhood  Neighborhood Latitude  Neighborhood Longitude  \
0    Wakefield              40.894705              -73.847201   
1    Wakefield              40.894705              -73.847201   
2    Wakefield              40.894705              -73.847201   
3    Wakefield              40.894705              -73.847201   
4    Wakefield              40.894705              -73.847201   

                         Venue  Venue Latitude  Venue Longitude  \
0             Lollipops Gelato       40.894123       -73.845892   
1           Ripe Kitchen & Bar       40.898152       -73.838875   
2              Ali's Roti Shop       40.894036       -73.856935   
3  Jackie's West Indian Bakery       40.889283       -73.843310   
4                     Rite Aid       40.896649       -73.844846   

         Venue Category  
0          Dessert Shop  
1  Caribbean Restaurant  
2  Caribbean Restaurant  
3  Caribbean Restaurant  
4              Pharmacy

In [44]:
super_city_venues.shape

(48362, 7)

In [45]:
super_city_venues['Venue Category'].unique()

array(['Dessert Shop', 'Caribbean Restaurant', 'Pharmacy', 'Burger Joint',
       'Bakery', 'Ice Cream Shop', 'Pizza Place', 'Seafood Restaurant',
       'Donut Shop', 'Southern / Soul Food Restaurant',
       'Mobile Phone Shop', 'Breakfast Spot', 'Sandwich Place',
       'Fried Chicken Joint', 'Fast Food Restaurant', 'Gas Station',
       'Supermarket', 'Nightclub', 'Rental Car Location', 'Bar',
       'Italian Restaurant', 'American Restaurant', 'Discount Store',
       'Food', 'Track', 'Bus Station', 'Metro Station', 'Cosmetics Shop',
       'Other Nightlife', 'Park', 'Train Station', 'Wine Shop',
       'Border Crossing', 'Chinese Restaurant', 'Dumpling Restaurant',
       'Shoe Store', 'Shopping Mall', 'Mexican Restaurant',
       'Spanish Restaurant', 'BBQ Joint', 'Coffee Shop', 'Bagel Shop',
       'Kids Store', 'Clothing Store', 'Lingerie Store', 'Diner', 'Spa',
       'Department Store', 'Video Game Store', 'Gym / Fitness Center',
       'Furniture / Home Store', 'Steakhouse'

In [48]:
print('There are {} uniques categories of venues in Super City.'.format(len(super_city_venues['Venue Category'].unique())))

There are 514 uniques categories of venues in Super City.


## Super City Neighborhood Analysis

In [128]:
# one hot encoding
super_city_onehot = pd.get_dummies(super_city_venues[['Venue Category']], prefix="", prefix_sep="")

# add neighborhood column back to dataframe
super_city_onehot['Neighborhood'] = super_city_venues['Neighborhood'] 

# set Neighborhood to index
cols_to_order = ['Neighborhood']
new_columns = cols_to_order + (super_city_onehot.columns.drop(cols_to_order).tolist())
super_city_onehot[new_columns]

super_city_onehot.shape

(48362, 514)

In [76]:
super_city_grouped = super_city_onehot.groupby('Neighborhood').mean().reset_index()
super_city_grouped

Neighborhood  ATM  Accessories Store  Adult Boutique  \
0            Acton Town Station  0.0               0.00             0.0   
1          Aldgate East Station  0.0               0.00             0.0   
2               Aldgate Station  0.0               0.00             0.0   
3            All Saints Station  0.0               0.00             0.0   
4                      Allerton  0.0               0.00             0.0   
5              Alperton Station  0.0               0.00             0.0   
6              Amersham Station  0.0               0.00             0.0   
7                 Angel Station  0.0               0.00             0.0   
8                      Annadale  0.0               0.00             0.0   
9               Archway Station  0.0               0.00             0.0   
10                Arden Heights  0.0               0.00             0.0   
11                    Arlington  0.0               0.00             0.0   
12          Arnos Grove Station  0.0               0.00             0.0   
13                     Arrochar  0.0               0.00             0.0   
14              Arsenal Station  0.0               0.00             0.0   
15                      Arverne  0.0               0.00             0.0   
16                      Astoria  0.0               0.00             0.0   
17              Astoria Heights  0.0               0.00             0.0   
18                   Auburndale  0.0               0.00             0.0   
19         Baker Street Station  0.0               0.00             0.0   
20               Balham Station  0.0               0.00             0.0   
21                 Bank Station  0.0               0.00             0.0   
22             Barbican Station  0.0               0.00             0.0   
23              Barking Station  0.0               0.00             0.0   
24          Barkingside Station  0.0               0.00             0.0   
25         Barons Court Station  0.0               0.00             0.0   
26                   Bath Beach  0.0               0.00             0.0   
27            Battery Park City  0.0               0.00             0.0   
28                    Bay Ridge  0.0               0.00             0.0   
29                  Bay Terrace  0.0               0.00             0.0   
..                          ...  ...                ...             ...   
574         West Harrow Station  0.0               0.00             0.0   
575     West India Quay Station  0.0               0.00             0.0   
576     West Kensington Station  0.0               0.00             0.0   
577        West Ruislip Station  0.0               0.00             0.0   
578                West Village  0.0               0.01             0.0   
579     Westbourne Park Station  0.0               0.00             0.0   
580          Westchester Square  0.0               0.00             0.0   
581                 Westerleigh  0.0               0.00             0.0   
582           Westferry Station  0.0               0.00             0.0   
583         Westminster Station  0.0               0.00             0.0   
584          White City Station  0.0               0.00             0.0   
585         Whitechapel Station  0.0               0.00             0.0   
586                  Whitestone  0.0               0.00             0.0   
587     Willesden Green Station  0.0               0.00             0.0   
588  Willesden Junction Station  0.0               0.00             0.0   
589              Williamsbridge  0.0               0.00             0.0   
590                Williamsburg  0.0               0.00             0.0   
591                 Willowbrook  0.0               0.00             0.0   
592      Wimbledon Park Station  0.0               0.00             0.0   
593           Wimbledon Station  0.0               0.00             0.0   
594             Windsor Terrace  0.0               0.00             0.0   
595                     Wingate  0.0        

Sort the venues in desceding order of presence

In [77]:
def return_most_common_venues(row, num_top_venues):
    row_categories = row.iloc[1:]
    row_categories_sorted = row_categories.sort_values(ascending=False)
    
    return row_categories_sorted.index.values[0:num_top_venues]

Display to 10 venue categories per neighborhood

In [161]:
num_top_venues = 10

indicators = ['st', 'nd', 'rd']

# create columns according to number of top venues
columns = ['Neighborhood']
for ind in np.arange(num_top_venues):
    try:
        columns.append('{}{} Most Common Venue'.format(ind+1, indicators[ind]))
    except:
        columns.append('{}th Most Common Venue'.format(ind+1))

# create a new dataframe
neighborhoods_venues_sorted = pd.DataFrame(columns=columns)
neighborhoods_venues_sorted['Neighborhood'] = super_city_grouped['Neighborhood']

for ind in np.arange(super_city_grouped.shape[0]):
    neighborhoods_venues_sorted.iloc[ind, 1:] = return_most_common_venues(super_city_grouped.iloc[ind, :], num_top_venues)

neighborhoods_venues_sorted

Neighborhood 1st Most Common Venue 2nd Most Common Venue  \
0            Acton Town Station                   Pub           Coffee Shop   
1          Aldgate East Station                 Hotel           Coffee Shop   
2               Aldgate Station                 Hotel           Coffee Shop   
3            All Saints Station           Coffee Shop                 Hotel   
4                      Allerton           Pizza Place            Donut Shop   
5              Alperton Station     Indian Restaurant         Grocery Store   
6              Amersham Station                   Pub            Restaurant   
7                 Angel Station                   Pub           Coffee Shop   
8                      Annadale    Italian Restaurant           Pizza Place   
9               Archway Station                   Pub                  Café   
10                Arden Heights           Pizza Place                  Park   
11                    Arlington        Discount Store  Fast Food Restaurant   
12          Arnos Grove Station         Grocery Store     Indian Restaurant   
13                     Arrochar    Italian Restaurant         Deli / Bodega   
14              Arsenal Station                  Café                   Pub   
15                      Arverne                 Beach             Surf Spot   
16                      Astoria      Greek Restaurant                   Bar   
17              Astoria Heights            Donut Shop                Bakery   
18                   Auburndale     Korean Restaurant           Pizza Place   
19         Baker Street Station                 Hotel          Burger Joint   
20               Balham Station                   Pub           Coffee Shop   
21                 Bank Station           Coffee Shop                 Hotel   
22             Barbican Station           Coffee Shop                   Pub   
23              Barking Station                 Hotel         Grocery Store   
24          Barkingside Station           Supermarket   Sporting Goods Shop   
25         Barons Court Station                   Pub           Pizza Place   
26                   Bath Beach    Italian Restaurant           Pizza Place   
27            Battery Park City                  Park           Coffee Shop   
28                    Bay Ridge           Pizza Place                   Spa   
29                  Bay Terrace              Bus Stop    Italian Restaurant   
..                          ...                   ...                   ...   
574         West Harrow Station           Coffee Shop         Grocery Store   
575     West India Quay Station           Coffee Shop                 Hotel   
576     West Kensington Station                   Pub                  Café   
577        West Ruislip Station                   Pub         Grocery Store   
578                West Village    Italian Restaurant                  Park   
579     Westbourne Park Station                   Pub                  Café   
580          Westchester Square           Pizza Place            Donut Shop   
581                 Westerleigh           Pizza Place     Convenience Store   
582           Westferry Station           Coffee Shop                  Park   
583         Westminster Station               Theater                 Hotel   
584          White City Station  Gym / Fitness Center                Bakery   
585         Whitechapel Station           Coffee Shop                  Café   
586                  Whitestone           Pizza Place         Deli / Bodega   
587     Willesden Green Station           Coffee Shop         Grocery Store   
588  Willesden Junction Station                   Pub           Coffee Shop   
589              Williamsbridge  Caribbean Restaurant           Pizza Place   
590                Williamsburg           Pizza Place   American Restaurant   
591                 Willowbrook           Pizza Place    Italian Restaurant   
592      Wimbledon Park Station          Tennis Court                

## Neighborhood Clustering

In [141]:
from sklearn.cluster import KMeans

# set number of clusters
kclusters = 10

super_city_grouped_clustering = super_city_grouped.drop('Neighborhood', 1)

# run k-means clustering
kmeans = KMeans(n_clusters=kclusters, random_state=0).fit(super_city_grouped_clustering)

# check cluster labels generated for each row in the dataframe
kmeans.labels_[0:10] 

array([1, 4, 4, 6, 2, 5, 9, 1, 3, 1], dtype=int32)

In [159]:
# add clustering labels
neighborhoods_venues_sorted.insert(0, 'Cluster Labels', kmeans.labels_)

super_city_merged = super_city

# merge toronto_grouped with toronto_data to add latitude/longitude for each neighborhood
super_city_merged = super_city_merged.join(neighborhoods_venues_sorted.set_index('Neighborhood'), on='Name')

super_city_merged.head() # check the last columns!

Name   Latitude  Longitude  Cluster Labels 1st Most Common Venue  \
0    Wakefield  40.894705 -73.847201               4  Caribbean Restaurant   
1   Co-op City  40.874294 -73.829939               4           Pizza Place   
2  Eastchester  40.887556 -73.827806               4  Caribbean Restaurant   
3    Fieldston  40.895437 -73.905643               0           Pizza Place   
4    Riverdale  40.890834 -73.912585               4           Pizza Place   

  2nd Most Common Venue 3rd Most Common Venue 4th Most Common Venue  \
0              Pharmacy           Pizza Place            Donut Shop   
1        Clothing Store  Fast Food Restaurant         Shopping Mall   
2  Fast Food Restaurant         Grocery Store            Donut Shop   
3          Burger Joint                   Bar    Mexican Restaurant   
4    Mexican Restaurant                  Park                   Bar   

  5th Most Common Venue 6th Most Common Venue 7th Most Common Venue  \
0                Bakery    Italian Restaurant  Fast Food Restaurant   
1  Caribbean Restaurant        Sandwich Place      Department Store   
2        Discount Store                 Diner         Shopping Mall   
3           Coffee Shop                  Park                  Bank   
4                  Bank                Bakery   Japanese Restaurant   

  8th Most Common Venue 9th Most Common Venue 10th Most Common Venue  
0     Mobile Phone Shop                   Bar    Rental Car Location  
1            Donut Shop     Mobile Phone Shop               Pharmacy  
2           Supermarket        Sandwich Place        Supplement Shop  
3    Italian Restaurant            Bagel Shop                 Bakery  
4        Sandwich Place            Donut Shop                  Diner

Visualize the clusters in each city

London

In [143]:
import folium
import matplotlib.cm as cm
import matplotlib.colors as colors

# London latitude and longitude
london_latitude=51.5074
london_longitude=0.1278

# create map
map_clusters = folium.Map(location=[london_latitude, london_longitude], zoom_start=11)

# set color scheme for the clusters
x = np.arange(kclusters)
ys = [i + x + (i*x)**2 for i in range(kclusters)]
colors_array = cm.rainbow(np.linspace(0, 1, len(ys)))
rainbow = [colors.rgb2hex(i) for i in colors_array]

# add markers to the map
markers_colors = []
for lat, lon, poi, cluster in zip(super_city_merged['Latitude'], super_city_merged['Longitude'], super_city_merged['Name'], super_city_merged['Cluster Labels']):
    label = folium.Popup(str(poi) + ' Cluster ' + str(cluster), parse_html=True)
    folium.CircleMarker(
        [lat, lon],
        radius=5,
        popup=label,
        color=rainbow[cluster-1],
        fill=True,
        fill_color=rainbow[cluster-1],
        fill_opacity=0.7).add_to(map_clusters)
       
map_clusters

In [144]:
# New York latitude and longitude
new_york_latitude=40.7128
new_york_longitude=-74.0060

# create map
map_clusters = folium.Map(location=[new_york_latitude, new_york_longitude], zoom_start=11)

# set color scheme for the clusters
x = np.arange(kclusters)
ys = [i + x + (i*x)**2 for i in range(kclusters)]
colors_array = cm.rainbow(np.linspace(0, 1, len(ys)))
rainbow = [colors.rgb2hex(i) for i in colors_array]

# add markers to the map
markers_colors = []
for lat, lon, poi, cluster in zip(super_city_merged['Latitude'], super_city_merged['Longitude'], super_city_merged['Name'], super_city_merged['Cluster Labels']):
    label = folium.Popup(str(poi) + ' Cluster ' + str(cluster), parse_html=True)
    folium.CircleMarker(
        [lat, lon],
        radius=5,
        popup=label,
        color=rainbow[cluster-1],
        fill=True,
        fill_color=rainbow[cluster-1],
        fill_opacity=0.7).add_to(map_clusters)
       
map_clusters

There are distinctive clusters across the two cities and only some partial overlapping.

## Cluster Examination

In [157]:
super_city_merged.loc[super_city_merged['Cluster Labels'] == 0, super_city_merged.columns[[0] + list(range(5, super_city_merged.shape[1]))]]


Name   Latitude  \
3                                     Fieldston  40.895437   
6                                   Marble Hill  40.876551   
12                                  City Island  40.847247   
19                                 High  Bridge  40.836623   
46                                    Bay Ridge  40.625801   
48                                  Sunset Park  40.645103   
49                                   Greenpoint  40.730201   
51                               Brighton Beach  40.576825   
54                                     Flatbush  40.636326   
55                                Crown Heights  40.670829   
57                                   Kensington  40.642382   
58                              Windsor Terrace  40.656946   
59                             Prospect Heights  40.676822   
61                                 Williamsburg  40.707144   
62                                     Bushwick  40.698116   
63                           Bedford Stuyvesant  40.687232   
64                             Brooklyn Heights  40.695864   
65                                  Cobble Hill  40.687920   
66                              Carroll Gardens  40.680540   
67                                     Red Hook  40.676253   
68                                      Gowanus  40.673931   
69                                  Fort Greene  40.688527   
70                                   Park Slope  40.672321   
78                                 Coney Island  40.574293   
81                                Dyker Heights  40.619219   
84                                 Clinton Hill  40.693229   
86                                     Downtown  40.690844   
87                                  Boerum Hill  40.685683   
88                    Prospect Lefferts Gardens  40.658420   
93                          Prospect Park South  40.647009   
..                                          ...        ...   
194                                North Corona  40.754071   
195                        Forest Hills Gardens  40.714611   
196                                  St. George  40.644982   
197                                New Brighton  40.640615   
219                                 Ditmas Park  40.643675   
236                                  Charleston  40.530531   
243                               Carnegie Hill  40.782683   
244                                        Noho  40.723259   
245                                Civic Center  40.715229   
246                               Midtown South  40.748510   
253                                Howland Hook  40.638433   
262                             Astoria Heights  40.770317   
264                           Concourse Village  40.824780   
268                               Hunters Point  40.743414   
269                                  Turtle Bay  40.752042   
270                                  Tudor City  40.746917   
271                             Stuyvesant Town  40.731000   
272                                    Flatiron  40.739673   
273                           Sunnyside Gardens  40.745652   
274                                  Blissville  40.737251   
275                                Fulton Ferry  40.703281   
276                                Vinegar Hill  40.703321   
277                                  Weeksville  40.675040   
279                                       Dumbo  40.703176   
385                       Goldhawk Road Station  51.501952   
403                        Holland Park Station  51.507334   
420                        Latimer Road Station  51.513544   
538                          White City Station  51.511978   
590             Shepherd's Bush Central Station  51.505561   
591  Shepherd's Bush Hammersmith & City Station  51.505561   

         2nd Most Common Venue      3rd Most Common Venue  \
3                 Burger Joint                        Bar   
6           Mexican Restaurant                       Café   
12             

In [146]:
super_city_merged.loc[super_city_merged['Cluster Labels'] == 1, super_city_merged.columns[[0] + list(range(5, super_city_merged.shape[1]))]]


Name 2nd Most Common Venue 3rd Most Common Venue  \
302           Acton Town Station           Coffee Shop  Gym / Fitness Center   
306             Amersham Station            Restaurant           Coffee Shop   
307                Angel Station           Coffee Shop                Bakery   
308              Archway Station                  Café    Italian Restaurant   
310              Arsenal Station                   Pub           Coffee Shop   
312               Balham Station           Coffee Shop           Pizza Place   
317         Barons Court Station           Pizza Place           Coffee Shop   
320         Belsize Park Station                  Café                Bakery   
321           Bermondsey Station               Brewery                   Pub   
322        Bethnal Green Station           Coffee Shop                  Café   
327         Boston Manor Station           Coffee Shop                   Gym   
329             Bow Road Station                  Café                  Park   
331              Brixton Station           Coffee Shop           Pizza Place   
332       Buckhurst Hill Station                  Café    Italian Restaurant   
334      Caledonian Road Station                  Café                  Park   
335          Camden Town Station           Coffee Shop           Music Venue   
336         Canada Water Station                  Park                  Café   
341   Chalfont & Latimer Station    Italian Restaurant                   Pub   
342           Chalk Farm Station                   Pub           Zoo Exhibit   
345              Chesham Station           Coffee Shop  Gym / Fitness Center   
346             Chigwell Station                   Pub            Restaurant   
347        Chiswick Park Station                  Café           Coffee Shop   
348          Chorleywood Station     Indian Restaurant                  Park   
349       Clapham Common Station                  Park           Coffee Shop   
350        Clapham North Station           Coffee Shop          Burger Joint   
351        Clapham South Station           Pizza Place           Coffee Shop   
352          Cockfosters Station      Greek Restaurant                  Café   
354        Colliers Wood Station         Grocery Store           Coffee Shop   
356              Croxley Station     Convenience Store                 Hotel   
361      Ealing Broadway Station                   Pub                  Park   
..                           ...                   ...                   ...   
532       West Hampstead Station                  Café                Bakery   
534      West Kensington Station                  Café           Pizza Place   
536      Westbourne Park Station                  Café                Bakery   
539          Whitechapel Station                  Café                   Pub   
541       Wimbledon Park Station                  Café           Coffee Shop   
543             Woodford Station                   Pub           Coffee Shop   
549           Bow Church Station                  Café           Art Gallery   
550       Bromley-By-Bow Station                  Café         Grocery Store   
553           Cutty Sark Station                Garden                  Café   
555      Deptford Bridge Station                  Café           Coffee Shop   
556          Devons Road Station           Coffee Shop         Grocery Store   
560        Elverson Road Station           Coffee Shop                  Café   
562            Greenwich Station                Garden                  Café   
563          Gunnersbury Station           Coffee Shop                  Park   
564          Hammersmith Station           Coffee Shop                  Café   
565  Hammersmith (D & P) Station           Coffee Shop                  Café   
571       Island Gardens Station                   Bar                Garden   
572         Kensal Green Station                  Café                  Park   
575          Kew Gardens Sta

In [147]:
super_city_merged.loc[super_city_merged['Cluster Labels'] == 2, super_city_merged.columns[[0] + list(range(5, super_city_merged.shape[1]))]]


Name 2nd Most Common Venue 3rd Most Common Venue  \
85         Sea Gate                  Park              Pharmacy   
170    Far Rockaway         Deli / Bodega           Pizza Place   
171   Broad Channel                 Beach    Chinese Restaurant   
172    Breezy Point                   Bar             Surf Spot   
176        Edgemere         Grocery Store            Donut Shop   
177         Arverne             Surf Spot         Metro Station   
178  Rockaway Beach           Pizza Place                   Bar   
179        Neponsit                  Park                   Spa   
189    Belle Harbor              Pharmacy           Pizza Place   
190   Rockaway Park           Pizza Place              Pharmacy   
191      Somerville             Surf Spot            Donut Shop   
284         Roxbury    Seafood Restaurant           Pizza Place   
298         Hammels           Pizza Place             Surf Spot   

    4th Most Common Venue       5th Most Common Venue 6th Most Common Venue  \
85            Supermarket  Construction & Landscaping          Tennis Court   
170                  Bank                  Donut Shop    Chinese Restaurant   
171              Dive Bar                  Playground        Sandwich Place   
172                 Diner               Grocery Store               Theater   
176           Pizza Place               Deli / Bodega            Skate Park   
177            Donut Shop              Sandwich Place           Pizza Place   
178        Ice Cream Shop                  Bagel Shop             Beach Bar   
179    Italian Restaurant                 Pizza Place            Donut Shop   
189            Donut Shop               Deli / Bodega        Ice Cream Shop   
190            Donut Shop                  Bagel Shop         Deli / Bodega   
191         Metro Station              Sandwich Place           Pizza Place   
284         Boat or Ferry                  Taco Place        Baseball Field   
298             Beach Bar               Deli / Bodega         Metro Station   

    7th Most Common Venue 8th Most Common Venue 9th Most Common Venue  \
85             Donut Shop    Chinese Restaurant      Basketball Court   
170                  Park   American Restaurant        Sandwich Place   
171       Other Nightlife           Pizza Place         Metro Station   
172   American Restaurant                  Park                 Trail   
176    Chinese Restaurant             Gift Shop                School   
177  Gym / Fitness Center                  Bank       Harbor / Marina   
178            Board Shop            Donut Shop           Supermarket   
179    Chinese Restaurant           Event Space              Boutique   
189                   Spa                  Bank            Board Shop   
190           Supermarket                  Bank                   Spa   
191  Caribbean Restaurant         Grocery Store       Thai Restaurant   
284           Art Gallery               Theater       Harbor / Marina   
298        Ice Cream Shop                   Bar    Chinese Restaurant   

    10th Most Common Venue  
85       Mobile Phone Shop  
170               Pharmacy  
171              Surf Spot  
172       Basketball Court  
176         Scenic Lookout  
177   Caribbean Restaurant  
178               Pharmacy  
179          Deli / Bodega  
189     Chinese Restaurant  
190         Ice Cream Shop  
191   Gym / Fitness Center  
284              BBQ Joint  
298               Wine Bar

In [148]:
super_city_merged.loc[super_city_merged['Cluster Labels'] == 3, super_city_merged.columns[[0] + list(range(5, super_city_merged.shape[1]))]]


Name      2nd Most Common Venue  \
305             Alperton Station              Grocery Store   
309          Arnos Grove Station          Indian Restaurant   
315              Barking Station              Grocery Store   
316          Barkingside Station        Sporting Goods Shop   
319            Becontree Station               Soccer Field   
324      Blackhorse Road Station                       Café   
328         Bounds Green Station                       Café   
330          Brent Cross Station                       Café   
333            Burnt Oak Station                       Park   
340          Canons Park Station                       Park   
353            Colindale Station       Gym / Fitness Center   
357        Dagenham East Station                        Bar   
358    Dagenham Heathway Station              Grocery Store   
359               Debden Station                Supermarket   
360          Dollis Hill Station              Grocery Store   
364           East Acton Station                Coffee Shop   
366             East Ham Station          Indian Restaurant   
368             Eastcote Station                        Pub   
369              Edgware Station                Coffee Shop   
371             Elm Park Station          Convenience Store   
376              Fairlop Station                        Pub   
378     Finchley Central Station                Supermarket   
382           Gants Hill Station                Pizza Place   
384        Golders Green Station              Grocery Store   
387          Grange Hill Station              Grocery Store   
391             Hainault Station                Supermarket   
393          Hanger Lane Station                      Hotel   
396       Hendon Central Station             Clothing Store   
406     Hounslow Central Station                      Hotel   
407        Hounslow East Station                   Bus Stop   
..                           ...                        ...   
478        Ruislip Manor Station                Coffee Shop   
480        Seven Sisters Station              Grocery Store   
485         South Harrow Station       Fast Food Restaurant   
487        South Ruislip Station                Gas Station   
494             Stanmore Station                       Park   
498         Sudbury Hill Station                        Pub   
499         Sudbury Town Station              Grocery Store   
505     Tooting Broadway Station              Grocery Store   
507       Tottenham Hale Station              Grocery Store   
513            Upminster Station                        Pub   
515                Upney Station                        Pub   
516           Upton Park Station              Grocery Store   
528           West Acton Station                Coffee Shop   
530        West Finchley Station         Turkish Restaurant   
533          West Harrow Station              Grocery Store   
535         West Ruislip Station              Grocery Store   
540      Willesden Green Station              Grocery Store   
542           Wood Green Station              Grocery Store   
544        Woodside Park Station              Grocery Store   
546              Beckton Station         Light Rail Station   
554               Cyprus Station                Coffee Shop   
561       Gallions Reach Station                Coffee Shop   
566            Harlesden Station  Middle Eastern Restaurant   
567  Harrow & Wealdstone Station          Indian Restaurant   
568   Harrow on the Hill Station                        Pub   
574               Kenton Station                        Pub   
582        North Wembley Station             Clothing Store   
592         South Kenton Station             Sandwich Place   
597     Stonebridge Park Station                      Hotel   
599      Wembley Central Station              Grocery Store   

    3rd Most Common Venue   4th Most Common Venue      5th Most Common Venue  \
305                 Hotel    Fast Food Re

In [149]:
super_city_merged.loc[super_city_merged['Cluster Labels'] == 4, super_city_merged.columns[[0] + list(range(5, super_city_merged.shape[1]))]]


Name      2nd Most Common Venue 3rd Most Common Venue  \
0              Wakefield                   Pharmacy           Pizza Place   
1             Co-op City             Clothing Store  Fast Food Restaurant   
2            Eastchester       Fast Food Restaurant         Grocery Store   
4              Riverdale         Mexican Restaurant                  Park   
5            Kingsbridge         Mexican Restaurant           Coffee Shop   
7               Woodlawn                   Pharmacy                   Bar   
8                Norwood       Caribbean Restaurant           Supermarket   
9         Williamsbridge                Pizza Place  Fast Food Restaurant   
10            Baychester                Bus Station           Pizza Place   
13          Bedford Park                      Diner                Garden   
14    University Heights  Latin American Restaurant    Spanish Restaurant   
15        Morris Heights             Sandwich Place  Fast Food Restaurant   
18            West Farms                Zoo Exhibit            Donut Shop   
20               Melrose             Discount Store                   Gym   
21            Mott Haven             Discount Store            Donut Shop   
22           Port Morris             Discount Store            Donut Shop   
23              Longwood             Discount Store              Pharmacy   
24           Hunts Point                Pizza Place  Fast Food Restaurant   
25            Morrisania                 Donut Shop           Pizza Place   
26             Soundview       Fast Food Restaurant                  Park   
27          Clason Point                Bus Station         Deli / Bodega   
30           Parkchester  Latin American Restaurant            Donut Shop   
31    Westchester Square                 Donut Shop                 Diner   
32              Van Nest                        Zoo           Zoo Exhibit   
33           Morris Park              Deli / Bodega           Coffee Shop   
35        Spuyten Duyvil         Mexican Restaurant                  Park   
36       North Riverdale                 Donut Shop                  Bank   
37            Pelham Bay                Pizza Place                Bakery   
40           Castle Hill                   Bus Stop           Bus Station   
41             Olinville       Caribbean Restaurant        Sandwich Place   
..                   ...                        ...                   ...   
204        Port Richmond             Ice Cream Shop     Convenience Store   
205     Mariner's Harbor           Department Store              Bus Stop   
208      New Springville                Coffee Shop        Cosmetics Shop   
217        Tompkinsville             Sandwich Place                   Bar   
220              Wingate                Pizza Place                Bakery   
221                Rugby                Pizza Place  Fast Food Restaurant   
224         Graniteville                Pizza Place    Italian Restaurant   
225            Arlington       Fast Food Restaurant    Spanish Restaurant   
234      Pleasant Plains                 Donut Shop           Pizza Place   
240    Heartland Village                Coffee Shop              Pharmacy   
254             Elm Park             Ice Cream Shop         Deli / Bodega   
255       Remsen Village                 Donut Shop           Pizza Place   
256             New Lots              Deli / Bodega  Fast Food Restaurant   
257      Paerdegat Basin       Caribbean Restaurant           Supermarket   
259        Jamaica Hills          Mobile Phone Shop              Pharmacy   
260               Utopia             Ice Cream Shop   Japanese Restaurant   
261              Pomonok                Bus Station                  Café   
263    Claremont Village                 Donut Shop         Grocery Store   
265           Mount Eden       Fast Food Restaurant            Donut Shop   
266           Mount Hope       Fast Food Restaurant         Grocery Store   
278    Broadway 

In [150]:
super_city_merged.loc[super_city_merged['Cluster Labels'] == 5, super_city_merged.columns[[0] + list(range(5, super_city_merged.shape[1]))]]


Name 2nd Most Common Venue  \
113                              Clinton                 Hotel   
114                              Midtown                 Hotel   
127                    Battery Park City           Coffee Shop   
139                     Long Island City                 Hotel   
209                               Travis                 Hotel   
241                           Bloomfield    Italian Restaurant   
267                         Sutton Place              Boutique   
297                         Hudson Yards           Art Gallery   
300                         Queensbridge           Coffee Shop   
303                      Aldgate Station           Coffee Shop   
304                 Aldgate East Station           Coffee Shop   
311                 Baker Street Station          Burger Joint   
313                         Bank Station                 Hotel   
314                     Barbican Station                   Pub   
318                    Bayswater Station           Coffee Shop   
323                  Blackfriars Station               Theater   
325                  Bond Street Station           Art Gallery   
326                      Borough Station    Italian Restaurant   
337                 Canary Wharf Station                 Hotel   
338                 Canning Town Station           Coffee Shop   
339                Cannon Street Station                 Hotel   
343                Chancery Lane Station                 Hotel   
344                Charing Cross Station        Ice Cream Shop   
355                Covent Garden Station                 Hotel   
363                 Earl's Court Station                 Hotel   
370            Elephant & Castle Station    Italian Restaurant   
372                   Embankment Station                 Hotel   
374                       Euston Station                  Café   
375                Euston Square Station                 Hotel   
377                   Farringdon Station                   Pub   
..                                   ...                   ...   
506         Tottenham Court Road Station           Coffee Shop   
509                   Tower Hill Station           Coffee Shop   
517                     Uxbridge Station    Italian Restaurant   
519                     Victoria Station                  Café   
523                Warren Street Station                 Hotel   
525                     Waterloo Station                 Hotel   
527                 Wembley Park Station                 Hotel   
537                  Westminster Station                 Hotel   
545                   All Saints Station                 Hotel   
547                 Beckton Park Station           Coffee Shop   
548                    Blackwall Station          Burger Joint   
551  Crossharbour & London Arena Station                  Park   
552                 Custom House Station         Grocery Store   
557                   East India Station           Coffee Shop   
558        Edgware Road Bakerloo Station                  Café   
559          Edgware Road Circle Station                  Café   
569   Heathrow Terminals 1, 2, 3 Station                 Hotel   
570                  Heron Quays Station  Gym / Fitness Center   
573         Kensington (Olympia) Station    Italian Restaurant   
576     King's Cross St. Pancras Station           Coffee Shop   
583                       Poplar Station                 Hotel   
584                Prince Regent Station           Coffee Shop   
588                 Royal Albert Station           Coffee Shop   
589               Royal Victoria Station           Coffee Shop   
593                   South Quay Station           Coffee Shop   
594             St. James's Park Station               Theater   
596                   St. Paul's Station                 Hotel   
598                Tower Gateway Station           Coffee Shop   
600              West India Quay Station                 Hotel   
601       

In [151]:
super_city_merged.loc[super_city_merged['Cluster Labels'] == 6, super_city_merged.columns[[0] + list(range(5, super_city_merged.shape[1]))]]


Name 2nd Most Common Venue 3rd Most Common Venue  \
11      Pelham Parkway            Donut Shop    Italian Restaurant   
16             Fordham           Pizza Place         Deli / Bodega   
17        East Tremont                   Zoo           Pizza Place   
28         Throgs Neck           Pizza Place                   Bar   
29        Country Club                Bakery        Sandwich Place   
34             Belmont           Pizza Place                Garden   
38       Schuylerville                   Bar   American Restaurant   
39      Edgewater Park           Pizza Place                   Bar   
47         Bensonhurst           Pizza Place                Bakery   
50           Gravesend              Pharmacy                Bakery   
52      Sheepshead Bay      Sushi Restaurant    Turkish Restaurant   
53   Manhattan Terrace            Bagel Shop      Sushi Restaurant   
77     Manhattan Beach    Turkish Restaurant        Sandwich Place   
79          Bath Beach           Pizza Place                  Bank   
83         Marine Park    Italian Restaurant        Clothing Store   
91        Bergen Beach           Pizza Place            Donut Shop   
98       Ocean Parkway      Sushi Restaurant            Bagel Shop   
133       Howard Beach        Ice Cream Shop           Pizza Place   
150         Whitestone         Deli / Bodega    Italian Restaurant   
153        Little Neck    Chinese Restaurant           Coffee Shop   
154         Douglaston           Coffee Shop    Chinese Restaurant   
161    Oakland Gardens          Intersection                 Trail   
174         Beechhurst            Donut Shop           Pizza Place   
175        Bay Terrace    Italian Restaurant                  Park   
186         Lindenwood        Ice Cream Shop            Donut Shop   
199           Rosebank        Sandwich Place    Italian Restaurant   
200      West Brighton           Pizza Place                  Park   
202          Todt Hill    Italian Restaurant           Pizza Place   
203        South Beach         Deli / Bodega                 Beach   
206         Port Ivory         Boat or Ferry          Intersection   
..                 ...                   ...                   ...   
223        Westerleigh     Convenience Store        Ice Cream Shop   
226           Arrochar         Deli / Bodega                  Park   
227           Grasmere              Pharmacy                  Bank   
228           Old Town           Pizza Place                  Bank   
229       Dongan Hills           Pizza Place                  Bank   
230      Midland Beach        Baseball Field     Convenience Store   
231         Grant City           Pizza Place              Pharmacy   
232     New Dorp Beach          Dessert Shop           Coffee Shop   
233           Huguenot         Train Station           Pizza Place   
235       Butler Manor                  Bank           Pizza Place   
237          Rossville           Pizza Place            Bagel Shop   
238      Arden Heights                  Park            Restaurant   
239         Greenridge      Sushi Restaurant        Ice Cream Shop   
242         Bulls Head    Chinese Restaurant                 Diner   
247      Richmond Town            Donut Shop                 Trail   
248        Shore Acres              Bus Stop         Deli / Bodega   
249            Clifton        Sandwich Place    Italian Restaurant   
250            Concord           Pizza Place              Bus Stop   
251       Emerson Hill           Pizza Place         Grocery Store   
252      Randall Manor                   Bar           Coffee Shop   
258         Mill Basin        Clothing Store    Italian Restaurant   
280      Manor Heights        Sandwich Place              Pharmacy   
281        Willowbrook    Italian Restaurant            Bagel Shop   
283        Egbertville           Pizza Place            Bagel Shop   
285          Homecrest            Bagel Shop           Pizza Place   
288    Lighthouse Hill    It